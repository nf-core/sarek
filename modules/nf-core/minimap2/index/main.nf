process MINIMAP2_INDEX {
    label 'process_low'

    // Note: the versions here need to match the versions used in minimap2/align
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/minimap2:2.28--he4a0461_0' :
        'biocontainers/minimap2:2.28--he4a0461_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.mmi"), emit: index
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    minimap2 \\
        -t $task.cpus \\
        -d ${fasta.baseName}.mmi \\
        $args \\
        $fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        minimap2: \$(minimap2 --version 2>&1)
    END_VERSIONS
    """

    stub:
    """
    touch ${fasta.baseName}.mmi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        minimap2: \$(minimap2 --version 2>&1)
    END_VERSIONS
    """
}