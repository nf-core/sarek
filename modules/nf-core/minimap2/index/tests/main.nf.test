nextflow_process {

    name "Test Process MINIMAP2_INDEX"
    script "../main.nf"
    process "MINIMAP2_INDEX"

    tag "modules"
    tag "modules_nfcore"
    tag "minimap2"
    tag "minimap2/index"

    test("minimap2 index") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}