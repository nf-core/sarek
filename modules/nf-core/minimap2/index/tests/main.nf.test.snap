{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_ref"
                        },
                        "genome.mmi:md5,72e450f12dc691e763c697463bdb1571"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0fced0ee8015e7f50b82566e3db8f7b0"
                ],
                "index": [
                    [
                        {
                            "id": "test_ref"
                        },
                        "genome.mmi:md5,72e450f12dc691e763c697463bdb1571"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0fced0ee8015e7f50b82566e3db8f7b0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-18T11:46:30.000058092"
    },
    "minimap2 index": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.mmi:md5,72e450f12dc691e763c697463bdb1571"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2f8340380c6741e9261a284262a90bde"
                ],
                "index": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.mmi:md5,72e450f12dc691e763c697463bdb1571"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2f8340380c6741e9261a284262a90bde"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T10:58:29.828187662"
    }
}